### PR TITLE
add Datree

### PR DESCRIPTION
**Pull request by hadar-co**
## Pre-requisites

- [x] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [x] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [x] Should include comments in the workflow for any parts that are not obvious or could use clarification.
- [x] Should specify least priviledge [permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) for `GITHUB_TOKEN` so that the workflow runs successfully.

**For _CI_ workflows, the workflow:**

- [ ] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [ ] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [ ] Packaging workflows should run on `release` with `types: [ created ]`.
- [ ] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [x] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/code-scanning).
- [x] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [x] `name`: Name of the Code Scanning integration.
  - [x] `creator`: Name of the organization/user producing the Code Scanning integration.
  - [x] `description`: Short description of the Code Scanning integration.
  - [x] `categories`: Array of languages supported by the Code Scanning integration.
  - [x] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [x] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [x] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [x] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [ ] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [ ] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

**Comment by hadar-co** [comment]
Hi @tiwarishub I see that this is still pending, anything else I need to do on my side?
Thanks

**Comment by Phantsure** [comment]
@hadar-co We are waiting for author verification. Apologies for the delay

**Comment by hadar-co** [comment]
@Phantsure Hi, is there an issue with the PR? :)

**Comment by mattlovestech** [comment]
@Phantsure @hadar-co has been verified as the owner, we're good to move forward.

File: code-scanning/datree.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+name: Datree
+
+on:
+  push:
+    branches: [ $default-branch, $protected-branches ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ $default-branch ]
+
```
Comment (Phantsure) [inline code-scanning/datree.yml:19]: Can we also have workflow level contents read permission?

File: code-scanning/datree.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  datree:
+    permissions:
+      contents: read # for actions/checkout to fetch code
+      security-events: write # for github/codeql-action/upload-sarif to upload SARIF results
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run Datree policy check
+        continue-on-error: true
+        uses: hadar-co/action-datree@main
```
Comment (Phantsure) [inline code-scanning/datree.yml:30]: Can we use this action instead of using personal account one? https://github.com/datreeio/action-datree

Comment (Phantsure) [reply code-scanning/datree.yml:30]: And please use commit sha as per security guidelines

File: code-scanning/datree.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        with:
+          # Add the path to the configuration file/s that you would like to test.
+          # See https://github.com/datreeio/action-datree#usage for all available options.
+          path: test-file.yaml
+          # Setting a SARIF output will generate a file named "datree.sarif" containing your test results
+          cliArguments: "-o sarif"
+      - name: Upload result to GitHub Code Scanning
+        uses: github/codeql-action/upload-sarif@v2
+        with:
+          sarif_file: datree.sarif
```
Comment (Phantsure) [inline code-scanning/datree.yml:44]:
```suggestion
          sarif_file: datree.sarif

```

Comment (Phantsure) [reply code-scanning/datree.yml:44]: To remove no new line at EOF

File: code-scanning/properties/datree.properties.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+    "name": "Datree",
+    "creator": "Datree",
+    "description": "Detect misconfigurations in your Kubernetes manifests and present them in Github code scanning",
+    "iconName": "datree",
+    "categories": ["Code Scanning", "YAML"]
+}
```
Comment (Phantsure) [inline code-scanning/properties/datree.properties.json:7]:
```suggestion
}

```

To remove no new line at EOF error

**Comment by Phantsure** [comment]
Left few comments @hadar-co. Rest looks good to me

**Comment by hadar-co** [comment]
@Phantsure Thanks, fixed the issues you mentioned:)

**Review by Phantsure — APPROVED** [review]
LGTM 🚀

**Comment by Phantsure** [comment]
Can you rebase/merge base in it?

File: code-scanning/datree.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run Datree policy check
+        continue-on-error: true
+        uses: datreeio/action-datree@de67ae7a5133d719dc794e1b75682cd4c5f94d8a
+        env:
+          # In order to use the Datree action you will need to have a Datree token.
+          # See https://hub.datree.io/setup/account-token#1-get-your-account-token-from-the-dashboard to acquire your token.
+          DATREE_TOKEN: ${{ secrets.DATREE_TOKEN }}
```
Comment (Phantsure) [inline code-scanning/datree.yml:37]:
```suggestion
          DATREE_TOKEN: ${{ secrets.DATREE_TOKEN }}
```

**Pull request merged**